data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java

```diff
@@ -120,4 +133,18 @@ public static OpensearchStateMachine createNew() {
     public StateMachineTracerAggregator getTracerAggregator() {
         return tracerAggregator;
     }
+
+    public void fire(OpensearchEvent trigger, OpensearchEvent errorEvent) {
```

Comment:
```suggestion
    private void fire(OpensearchEvent trigger, OpensearchEvent errorEvent) {
```

This can be private to avoid confusion.

Reply:
you are right. I only wanted to provide this if we would like to set a different errorEvent at some point in time